geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -221,17 +221,18 @@ public long lpush(ExecutionHandlerContext context, List<byte[]> elementsToAdd,
    */
   public int lrem(int count, byte[] element, Region<RedisKey, RedisData> region, RedisKey key) {
     List<Integer> removedIndexes;
-    byte version;
+    byte newVersion;
     synchronized (this) {
       removedIndexes = elementList.remove(element, count);
-      version = incrementAndGetVersion();
-    }
+      if (removedIndexes.isEmpty()) {
+        return 0;
+      }
 
-    if (!removedIndexes.isEmpty()) {
-      storeChanges(region, key,
-          new RemoveElementsByIndex(version, removedIndexes));
+      newVersion = incrementAndGetVersion();
```

Comment:
This change is not part of implementing RPUSHX and shouldn't really be included here. Fixing multiple unrelated issues in one PR can make tracking down the cause of bugs difficult, and cause trouble if things need to be reverted. If there's a bug that's being fixed with this change, a separate PR should be opened for it.

Reply:
I talked to Wayne and he said typically we would, but since I'm going on vacation soon he said it's ok to put this fix in here. If it'll be better in the long run to create a ticket, I'll do so and take this out.